delta/plugins/blazegraph/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/blazegraph/indexing/BlazegraphIndexingCoordinator.scala

```diff
@@ -0,0 +1,181 @@
+      views: BlazegraphViews,
+      eventLog: GlobalEventLog[Message[ResourceF[Graph]]],
+      client: BlazegraphClient,
+      projection: Projection[Unit],
+      config: BlazegraphViewsConfig
+  )(implicit
+      as: ActorSystem[Nothing],
+      scheduler: Scheduler,
+      base: BaseUri,
+      lens: ViewLens[ResourceF[IndexingBlazegraphView]]
+  ): Task[IndexingStreamCoordinator[ResourceF[IndexingBlazegraphView]]] = {
+
+    def buildStream(
+        viewRes: ResourceF[IndexingBlazegraphView],
+        initialProgress: ProjectionProgress[Unit]
+    ): Task[Stream[Task, Unit]] = {
+      val view                                = viewRes.value
+      implicit val projectionId: ProjectionId = viewRes.projectionId
+      val namespace                           = s"${config.client.indexPrefix}_${view.uuid}_${viewRes.rev}"
+      val retryStrategy                       = RetryStrategy[SparqlClientError](
+        config.client.retry,
+        {
+          case WrappedHttpClientError(_) => true
+          case _                         => false
+        },
+        logError(logger, "blazegraph client")
+      )
+      import retryStrategy.{errorHandler, policy, retryWhen}
+      for {
+        _     <- client
+                   .createNamespace(namespace, indexProperties)
+                   .retryingOnSomeErrors(retryWhen)
+                   .hideErrorsWith(err => new IllegalArgumentException(err.toString()))
+        eLog  <- eventLog
+                   .stream(view.project, initialProgress.offset, view.resourceTag)
+                   .hideErrorsWith(e => new IllegalArgumentException(e.reason))
+        stream = eLog
+                   .filterMessage(res =>
+                     (view.resourceTypes.isEmpty || res.types.exists(view.resourceTypes.contains))
+                       && (view.resourceSchemas.isEmpty || view.resourceSchemas.contains(
+                         res.schema.iri
+                       )) && (!res.deprecated || view.includeDeprecated)
+                   )
+                   .flatMapMessage { msg =>
+                     val graph = if (view.includeMetadata) {
+                       val types: Set[Triple] = msg.value.types.map((msg.value.id, rdf.tpe, _))
+                       msg.value.value
+                         .add(types)
+                         .add(nxv.rev.iri, msg.value.rev)
+                         .add(nxv.deprecated.iri, msg.value.deprecated)
+                         .add(nxv.createdAt.iri, msg.value.createdAt)
+                         .add(nxv.updatedAt.iri, msg.value.updatedAt)
+                         .add(nxv.updatedBy.iri, msg.value.updatedBy.id)
+                         .add(nxv.createdBy.iri, msg.value.createdBy.id)
+                         .add(nxv.schemaId.iri, msg.value.schema.iri)
+                         .add(nxv.project.iri, ResourceUris.project(view.project).accessUri)
+                         .add(nxv.incoming.iri, UriUtils./(msg.value.uris.accessUri, "incoming"))
```

Comment:
use the syntax instead:

```scala
import ch.epfl.bluebrain.nexus.delta.rdf.syntax._

.add(nxv.incoming.iri, msg.value.uris.accessUri / "incoming")
.add(nxv.outgoing.iri, msg.value.uris.accessUri / "outgoing")

```

Reply:
I tried that, but for some reason this refuses to compile.